pkg/machinepool/machinepool_test.go

```diff
@@ -307,5 +352,179 @@ var _ = Describe("Machinepool and nodepool", func() {
 			Expect(MachinePoolKeyRE.MatchString("m123123123123123123123123123")).To(BeTrue())
 			Expect(MachinePoolKeyRE.MatchString("m#123")).To(BeFalse())
 		})
+		It("Tests getMachinePoolAvailabilityZones", func() {
```

Comment:
This is in the context of `editMachinePoolAutoscaling` is this the correct place? 

IMO create a new `Context` for testing `getMachinePoolAvailabilityZones`